iml-gui/crate/src/components/activity_indicator.rs

```diff
@@ -0,0 +1,49 @@
+use crate::{components::font_awesome, generated::css_classes::C};
+use iml_wire_types::{Alert, AlertSeverity};
+use seed::{prelude::*, *};
+use std::collections::{HashMap, HashSet};
+
+pub fn update_activity_health(
+    active_alert: &HashMap<u32, Alert>,
+) -> ActivityHealth {
+    let xs = active_alert
+        .iter()
+        .filter(|(_, x)| match x.severity {
+            AlertSeverity::WARNING | AlertSeverity::ERROR => true,
+            _ => false,
+        })
+        .map(|(_, x)| &x.severity)
+        .collect::<Vec<_>>();
+
+    let count = xs.len();
+
+    let s = xs.into_iter().collect::<HashSet<&AlertSeverity>>();
+
+    let mut health = AlertSeverity::INFO;
+
+    if s.contains(&AlertSeverity::ERROR) {
+        health = AlertSeverity::ERROR
+    } else if s.contains(&AlertSeverity::WARNING) {
+        health = AlertSeverity::WARNING;
+    }
+
+    ActivityHealth {
+        health,
+        count,
+    }
+}
```

Comment:
I just performed some Rust exercise:

If  we had this definition of  `AlertSeverity`:
```rust
#[derive(
    serde::Serialize, serde::Deserialize, Clone, Copy, Debug, Hash, PartialEq, Eq, PartialOrd, Ord,
)]
pub enum AlertSeverity {
    DEBUG,
    INFO,
    WARNING,
    ERROR,
    CRITICAL,
}
```

then we could have it much shorter (and probably faster, at least with just a single pass):
```rust
pub fn update_activity_health(
    active_alert: &HashMap<u32, Alert>,
) -> ActivityHealth {
    let (health, count) = active_alert
        .iter()
        .filter(|(_, x)| x.severity > AlertSeverity::INFO)
        .fold((AlertSeverity::INFO, 0), |(h, c), (_, x)| {
            (max(h, x.severity), c + 1)
        });

    ActivityHealth {
        health,
        count,
    }
}
```

Reply:
Sure, I'll update. We may as well just fold over `ActivityHealth` directly.